### DISCUSSION TITLE
我的 PKM 系统

### DISCUSSION THREAD

**Discussion by giscus** [edony-ink/comments#8] [General]
# 我的 PKM 系统

基于 Obsidian 构建的个人 PKM 系统。

https://www.edony.ink/my-pkm/

**Comment by edonyzpc** [comment]
PKM 系统已经是上一个版本了，后面会 update 一下，涉及到：
- 插件 list
- minimal theme 5.3.0 配置
- shortcuts of obsidian
- automation of obsidian

**Comment by hinzer** [comment]
请教博主，您的"index notes"是怎么记的？看你在文中介绍像是图表，是一个脑图或者概念图的形式吗，然后unsorted或者fleeting是这个图表中部分的进一步阐述？

**Reply by edonyzpc** [reply]
我用截图演示一下我平时跟 index 相关的工作流：
![image](https://user-images.githubusercontent.com/7744664/176347438-1541a501-9d3f-4575-aa0a-75d385e85b7a.png)

**Reply by edonyzpc** [reply]
上图所示，我是以 technology 主题作为演示的，其他主题工作流是类似的。需要更多细节可以再问我

**Reply by hinzer** [reply]
谢谢。看上去很不错，有机会我也试试。哈哈

**Reply by hinzer** [reply]
最左边是目录树吧，"technology"和"index-technology"应该表示层级关系，为什么要设置到一个层级，看上去很奇怪

**Reply by hinzer** [reply]
还有一个问题，我很好奇右边的链接图怎么生成的。看上去中间那个点点是主题，周围是子主题么？这是obsidian根据层级关系自动生成的么

**Reply by edonyzpc** [reply]
> 最左边是目录树吧，"technology"和"index-technology"应该表示层级关系，为什么要设置到一个层级，看上去很奇怪

这个是个人喜好问题，完全可以按照自己的理解来安排。我自己的理解是：index 目录属于主题笔记的索引，index-{{subject}} 目录表示 {{subject}} 主题的子主题集合，e.g. index-technology 表示 technology 主题下的子主题集合

**Reply by edonyzpc** [reply]
> 还有一个问题，我很好奇右边的链接图怎么生成的。看上去中间那个点点是主题，周围是子主题么？这是obsidian根据层级关系自动生成的么

是的，这个链接图是 obsidian 自动生成的，我是为让你更加清楚看出 index 目录的结构，加了 Filter：`path:(1.index) OR (1.index/index-technology))`

可以参考 Obsidian help：
- [obsidian search syntax](https://help.obsidian.md/Plugins/Search#Search+operators)
- [obsidian graph view](https://help.obsidian.md/Plugins/Graph+view)

**Reply by hinzer** [reply]
您好，在请教个问题：如果将不同目录(层级)下包含了相同的子目录(主题)，在index目录下也能根据主题生成一个完全的索引么？比如我有日常记录的习惯，这个"日常记录"我希望放在一个层级下面，然后根据领域划分到不同的主题(工作啊生活之类的)。
不确定您的PKM系统里有没有包含这个功能，如果有的话，是如何维护index目录里的内容的呢？

**Reply by edonyzpc** [reply]
### 需求理解
先理解一下你的需求（看看是否OK）：你是希望自己写的一些笔记能够自动索引到 index 目录或者子目录下。

### 效果展示
我也有类似的场景，我在写日记的时候，会有一个 index-diary 专门做日记的索引，你给你看一下效果图：
![image](https://user-images.githubusercontent.com/7744664/178641487-bd5a5080-5421-4681-bc1e-930ad1881139.png)

### 实现步骤
具体的实现方式时借助 templater 插件自己写的脚本：
1. 编写 templater 模版（包括脚本），模版放到 templater 插件配置的模版目录中；
2. 自定义 templater 插件快捷键配置，我的是 shift + cmd + 2（这个是方便通过快捷键添加索引）；
![image](https://user-images.githubusercontent.com/7744664/178641169-37a3d6e1-ca9d-4f3f-bcc4-15c9aa62b7ac.png)
3. 配置 diary notes，自动触发 templater 执行
![image](https://user-images.githubusercontent.com/7744664/178640848-0d2e8fe9-9751-4cde-b99a-349600c1b4a6.png)
4. 点击对应日期自动创建 diary 文档，同时自动执行 templater 模版，这样就自动添加到 index 索引了
![image](https://user-images.githubusercontent.com/7744664/178641070-c4c4bf5a-9988-44be-94cb-a39339e15602.png)

### templater 教程
**P.S. templater 使用我有一个教程可以参考**：[Obsidian 自动添加元数据](https://www.edony.ink/obsidian-auto-metadata/)

**Reply by hinzer** [reply]
谢谢，我大概明白您的意思了：通过templater自动创建日记模板，其中定义了一些元数据包含主题标签，索引页里面展示了是这个主题标签的所有文件。
 index-diary 这个目录下面的索引页您平时是手工维护的吗，有没有配置将一个标签和索引页自动关联，填充的功能？

**Reply by hinzer** [reply]
试了下，按照这个方法可以自动添加啊。GOOD！！

**Reply by edonyzpc** [reply]
我是通过两个标签在顶层 index 中进行自动关联的，两个标签是：
1. `#005-diary`
2. `#{yeary}`, (e.g. `#2022`)

在顶层 index 自动索引的发方式是用 dateview：
![image](https://user-images.githubusercontent.com/7744664/178669206-b36623a0-b6a8-4a3f-9ec6-a3ddb3cc8fa4.png)

**Comment by hinzer** [comment]
您好老哥。不知我理解index目录 是不是这样：index 目录下是笔记的索引，比如拿technology这个主题为例，一类索引是在 technology 文件中以 目录树 的形式组织的；另外一类是 index-technology 目录下面的文件，作为子主题。

然后我想问个问题：
1、 index-technology 目录和下面的文件是基于什么规则关联的？（不太清楚 链接图 中主题和子主题之间的链接关系）
2、 index-technology 下面的文件内容是怎么维护的？子主题需要在链接其他文件吗

**Reply by edonyzpc** [reply]
主题和子主题我是手动维护的，按照自己的理解和需要，比如 technology 子主题我是这么分的：
![image](https://user-images.githubusercontent.com/7744664/179142867-bdbc0335-8f47-4f53-b782-22af7cf88c90.png)

子主题之间是无序的，因为 zettelkasten 的理念是为了能够发散思考都是双链关联上的，不把他们按照一定的逻辑结构化。子主题的维护主要是通过 `#tag`，关键字进行自动关联的，也有一些是 alias 或者文件名我手动关联上的，这些笔记文件的链接都是我在回顾的时候逐步丰富的，backlink、outlink 点点就可以了。

**Reply by hinzer** [reply]
嗯嗯，谢谢！GOOD FOR ME

**Comment by hinzer** [comment]
我最近在尝试把笔记迁入obsidian上，想按照文中的结构设置。可以发我一个demo版本吗 :)

**Reply by edonyzpc** [reply]
可以，不过要周末才行。我这周末看看搞一个模版出来放到 github 上，后面我把项目地址发出来。

**Reply by hinzer** [reply]
记得在这里留个链接，嘿嘿 :)

**Reply by edonyzpc** [reply]
https://github.com/edony-shadow-walker/pkm-template
